arangod/Pregel/Conductor.cpp

```diff
@@ -116,7 +119,7 @@ Conductor::Conductor(
                                    "Algorithm not found");
   }
   _masterContext.reset(_algorithm->masterContext(config));
-  _aggregators = std::make_unique<AggregatorHandler>(_algorithm.get());
+  _aggregators = std::make_shared<AggregatorHandler>(_algorithm.get());
```

Comment:
I had to make the AggregatorHandler shared to be able to use the inspector on the ContinueRecovery message, which includes an AggregatorHandler. Do you have a better idea how to do this?

Reply:
You could change the wrapper struct use a (const?) reference to the aggregator handler. In this case this is "fine" because we know that the aggregator handler will live longer than the serialisation function needs it.

That said, I do not quite understand why `_aggregators` is a `unique_ptr` at all; it just irked me that it got changed ;) I think a `shared_ptr` is just fine (and safe).

We will change this code again soon, anyway ;).